### PR TITLE
fix: Adding back endpoint removed by mistake [DHIS2-13006]

### PR DESCRIPTION
**Pull request by maikelarabori**
**Needs formal approval from Release Control Board.**

### CODE REVIEW

**Comment by larshelge** [comment]
Should we just allow all analytics requests for XML? Including `/analytics` ? I think that is better, as some might be using the XML there too.

**Comment by larshelge** [comment]
We should also include API versions like for the other.

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/mvc/messageconverter/XmlPathMappingJackson2XmlHttpMessageConverter.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
         Pattern.compile( "/(\\d\\d/)?events(/?$|/.+)" ),
         Pattern.compile( "/(\\d\\d/)?trackedEntityInstances(/?$|/.+)" ),
         Pattern.compile( "/(\\d\\d/)?dataValueSets(/?$|/.+)" ),
+        Pattern.compile( "/analytics/dataValueSet.xml" ),
```
Comment (Philip-Larsen-Donnelly) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/mvc/messageconverter/XmlPathMappingJackson2XmlHttpMessageConverter.java:55]: Will this handle versioned endpoints without the (\\d\\d/)? pattern.
In general, shouldn't it align with the others?
e.g.
"/(\\d\\d/)?dataValueSet(/?$|/.+)"
?

Comment (Philip-Larsen-Donnelly) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/mvc/messageconverter/XmlPathMappingJackson2XmlHttpMessageConverter.java:55]: Sorry - I see Lars already made this comment.

Comment (maikelarabori) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/mvc/messageconverter/XmlPathMappingJackson2XmlHttpMessageConverter.java]: Done. Versions included.

**Comment by maikelarabori** [comment]
> Should we just allow all analytics requests for XML? Including `/analytics` ? I think that is better, as some might be using the XML there too.

Done, using `/analytics` as regexp.

**Comment by maikelarabori** [comment]
> We should also include API versions like for the other. The `/(\\d\\d/)?` refers to the optional API version @maikelarabori .

Done, versions included.

**Review by larshelge — APPROVED** [review]

**Pull request merged**